modules/replicator/src/main/java/org/apache/ignite/internal/replicator/message/PrimaryReplicaRequest.java

```diff
@@ -21,4 +21,11 @@
  * This is an interface for requests are designed for primary replica.
  */
 public interface PrimaryReplicaRequest extends ReplicaRequest {
+    /**
+     * Gets an enlistment consistency token.
+     * The token is used to check that the lease is still actual while the message goes to the replica.
+     *
+     * @return Enlistment consistency token.
+     */
+    long enlistmentConsistencyToken();
```

Comment:
Why it's long and not Long? Originally it was marked as @Nullable and by the fact that you've added an assert within `ensureReplicaIsPrimary` I believe that it's no longer true. Thus I'd rather have it Long without @Nullable - in that case we will check whether proper value is propagated to a message or not.

Reply:
Look at the comment above.